sidecar/tlv.go

```diff
@@ -0,0 +1,377 @@
+package sidecar
+
+import (
+	"io"
+
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/btcsuite/btcutil"
+	"github.com/lightninglabs/pool/order"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/tlv"
+)
+
+const (
+	idType        tlv.Type = 1
```

Comment:
any given reason for using odd numbers here? Since it is a custom type.

Reply:
No, I just copied the pattern I found in https://github.com/lightningnetwork/lnd/blob/master/channeldb/invoices.go#L129.
I probably need to read up on the implication of even/odd types. Or does this only apply to the LN gossip wire protocol?

Reply:
Ah, I think I get it now. If a type is _unknown_ and is odd, the tlv library will just ignore it. But since we know all our types here, we don't need to use odd types. Fixed.